Dockerfile

```diff
@@ -1,10 +1,17 @@
 ARG ROS_VERSION=galactic
 FROM ghcr.io/aica-technology/ros2-control-libraries:${ROS_VERSION} as development
 
+# upgrade ament_cmake_python
+RUN sudo apt update && sudo apt install -y ros-${ROS_DISTRO}-ament-cmake-python && sudo rm -rf /var/lib/apt/lists/*
+
 WORKDIR ${HOME}/ros2_ws
 # copy sources and build ROS workspace with user permissions
 WORKDIR ${HOME}/ros2_ws/
-COPY --chown=${USER} ./source/ ./src/
+COPY --chown=${USER} ./source/modulo_core ./src/modulo_core
+# build modulo
+RUN /bin/bash -c "source /opt/ros/$ROS_DISTRO/setup.bash; colcon build"
```

Comment:
I assume you build the core first so that you can develop the components (with core dependencies) without having to compile the components package (which might be broken during development). However it's not ideal that the development layer builds one package and not the other.

For cleanness and flexibility, I would recommend you split this into multiple layers, something along the lines of:

```dockerfile
FROM ros2-control-libraries as dependencies
WORKDIR ${HOME}/ros2_ws
RUN sudo apt update && sudo apt install -y ros-${ROS_DISTRO}-ament-cmake-python && sudo rm -rf /var/lib/apt/lists/*

FROM dependencies as modulo-core
COPY --chown=${USER} ./source/modulo_core ./src/modulo_core
RUN /bin/bash -c "source /opt/ros/$ROS_DISTRO/setup.bash; colcon build --packages-select modulo_core"

FROM modulo-core as modulo-components
COPY --chown=${USER} ./source/modulo_components ./src/modulo_components
RUN /bin/bash -c "source /opt/ros/$ROS_DISTRO/setup.bash; colcon build --packages-select modulo_components"
```